src/characteristic.c

```diff
@@ -5,12 +5,60 @@
  *
  **********************************************************************/
 #include <stdlib.h>
+#include <stdio.h>
 #include <stdbool.h>
 #include <string.h>
 
 #include "characteristic.h"
-
-characteristic_t *characteristic_new (const char *uuid, const char *service_path, descriptor_t *descriptors)
+#include "descriptor.h"
+#include "dbusutils.h"
+#include "defines.h"
+#include "utils.h"
+
+static void characteristic_handle_unregister_device (DBusConnection *connection, void *data);
+static DBusHandlerResult characteristic_handle_dbus_message (DBusConnection *connection, DBusMessage *message, void *data);
+static void characteristic_get_uuid(void *user_data, DBusMessageIter* iter);
+static void characteristic_get_service(void *user_data, DBusMessageIter* iter);
+static void characteristic_get_flags(void *user_data, DBusMessageIter* iter);
+static void characteristic_get_value(void *user_data, DBusMessageIter* iter);
+static void characteristic_read_value (characteristic_t *characteristic, DBusMessageIter* iter);
+
+DBusObjectPathVTable characteristic_dbus_callbacks = {
+  .unregister_function = characteristic_handle_unregister_device,
+  .message_function = characteristic_handle_dbus_message,
+};
+
+static dbus_property_t characteristic_properties[] =
+{
+  {BLE_PROPERTY_UUID, DBUS_TYPE_STRING_AS_STRING, characteristic_get_uuid}, //s
```

Comment:
Comments on this line and the following ones should either be expanded to be meaningful, removed or reworked as a comment block for characteristic_properties to describe whats going on.